src/js/stores/MesosLogStore.js

```diff
@@ -1,30 +1,62 @@
-import {Store} from 'mesosphere-shared-reactjs';
+import BaseStore from './BaseStore';
 
 import ActionTypes from '../constants/ActionTypes';
 import AppDispatcher from '../events/AppDispatcher';
 import Config from '../config/Config';
-import EventTypes from '../constants/EventTypes';
-import GetSetMixin from '../mixins/GetSetMixin';
+import EventTypes from '../constants/EventTypes';;
 import Item from '../structs/Item';
 import LogBuffer from '../structs/LogBuffer';
 import MesosLogActions from '../events/MesosLogActions';
 
 const MAX_FILE_SIZE = 50000;
 
-const MesosLogStore = Store.createStore({
-  storeID: 'mesosLog',
-
-  mixins: [GetSetMixin],
-
-  addChangeListener: function (eventName, callback) {
+class MesosLogStore extends BaseStore {
+
+  constructor() {
+    super(...arguments);
+
+    this.dispatcherIndex = AppDispatcher.register((payload) => {
```

Comment:
If we can make sure that one can only create a single instance of this class, then there is no need to move the registering part. The current "singleton" pattern is the cheapest of all available patterns but we were using it already (e.g. [CompositeState](https://github.com/dcos/dcos-ui/blob/master/src/js/structs/CompositeState.js#L103)).

Let's add this topic to our agenda and decide on a true and safe singleton pattern and refactor the store accordingly. I'd favor a pattern that allows us to extend singleton classes and one that is not only relying on "proper" file handling (e.g. case sensitivity).